src/UsgsAstroSarSensorModel.cpp

```diff
@@ -513,47 +635,64 @@ csm::ImageCoord UsgsAstroSarSensorModel::getImageStart() const
 
 csm::ImageVector UsgsAstroSarSensorModel::getImageSize() const
 {
-  return csm::ImageVector(0.0, 0.0);
+  return csm::ImageVector(m_nLines, m_nSamples);
 }
 
 pair<csm::ImageCoord, csm::ImageCoord> UsgsAstroSarSensorModel::getValidImageRange() const
 {
-  return make_pair(csm::ImageCoord(0.0, 0.0), csm::ImageCoord(0.0, 0.0));
+  csm::ImageCoord start = getImageStart();
+  csm::ImageVector size = getImageSize();
+  return make_pair(start, csm::ImageCoord(start.line + size.line, start.samp + start.samp));
 }
 
 pair<double, double> UsgsAstroSarSensorModel::getValidHeightRange() const
 {
-  return make_pair(0.0, 0.0);
+  return make_pair(m_minElevation, m_maxElevation);
 }
 
 csm::EcefVector UsgsAstroSarSensorModel::getIlluminationDirection(const csm::EcefCoord& groundPt) const
 {
-  return csm::EcefVector(0.0, 0.0, 0.0);
+  csm::EcefVector sunPosition = getSunPosition(getImageTime(groundToImage(groundPt)));
+  csm::EcefVector illuminationDirection = csm::EcefVector(groundPt.x - sunPosition.x,
+                                                          groundPt.y - sunPosition.y,
+                                                          groundPt.z - sunPosition.z);
+
+  double scale = sqrt(illuminationDirection.x * illuminationDirection.x +
+                      illuminationDirection.y * illuminationDirection.y +
+                      illuminationDirection.z * illuminationDirection.z);
+
+  illuminationDirection.x /= scale;
+  illuminationDirection.y /= scale;
+  illuminationDirection.z /= scale;
+  return illuminationDirection;
 }
 
 double UsgsAstroSarSensorModel::getImageTime(const csm::ImageCoord& imagePt) const
 {
-  return 0.0;
+  return m_startingEphemerisTime + (imagePt.line - 0.5) * m_exposureDuration;
 }
 
 csm::EcefCoord UsgsAstroSarSensorModel::getSensorPosition(const csm::ImageCoord& imagePt) const
 {
-  return csm::EcefCoord(0.0, 0.0, 0.0);
+  double time = getImageTime(imagePt);
+  return getSensorPosition(time);
 }
 
 csm::EcefCoord UsgsAstroSarSensorModel::getSensorPosition(double time) const
 {
-  return csm::EcefCoord(0.0, 0.0, 0.0);
+  csm::EcefVector sensorVector = getSpacecraftPosition(time);
+  return csm::EcefCoord(sensorVector.x, sensorVector.y, sensorVector.z);
 }
 
 csm::EcefVector UsgsAstroSarSensorModel::getSensorVelocity(const csm::ImageCoord& imagePt) const
 {
-  return csm::EcefVector(0.0, 0.0, 0.0);
+  double time = getImageTime(imagePt);
+  return getSensorVelocity(time);
 }
 
 csm::EcefVector UsgsAstroSarSensorModel::getSensorVelocity(double time) const
 {
-  return csm::EcefVector(0.0, 0.0, 0.0);
+  return getSpacecraftVelocity(time);
```

Comment:
Is the separate `getSpacecraftVelocity` function required by the API, or can we just delete it and replace it with `getSensorVelocity`?

Reply:
The getSpacecraft* functions are not part of the API and I think we can just axe them in favor of the getSensor* functions which are part of the API.